python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -208,6 +221,228 @@ def columns_in_target(self) -> Set[str]:
+        doc='Ceiling (maximum/faint) magnitude for target sources',
+    )
+    mag_faintest_ref = pexConfig.Field(
+        dtype=float,
+        default=30,
+        doc='Faintest magnitude cutoff for binning',
+    )
+    mag_zeropoint_ref = pexConfig.Field(
+        dtype=float,
+        default=31.4,
+        doc='Magnitude zeropoint for reference sources',
+    )
+    mag_zeropoint_target = pexConfig.Field(
+        dtype=float,
+        default=31.4,
+        doc='Magnitude zeropoint for target sources',
+    )
+
+
+class Measurement(Enum):
+    DIFF = auto()
+    CHI = auto()
+
+
+class Statistic(metaclass=ABCMeta):
+    """A statistic that can be applied to a set of values.
+    """
+    @abstractmethod
+    def value(self, values):
+        """Return the value of the statistic given a set of values.
+
+        Parameters
+        ----------
+        values : `Collection` [`float`]
+            A set of values to compute the statistic for.
+        Returns
+        -------
+        statistic : `float`
+            The value of the statistic.
+        """
+        pass
+
+
+class Median(Statistic):
+    """The median of a set of values."""
+    def value(self, values):
+        return np.median(values)
+
+
+class SigmaIQR(Statistic):
+    """The re-scaled inter-quartile range (sigma equivalent)."""
+    def value(self, values):
+        return iqr(values, scale='normal')
+
+
+class SigmaMAD(Statistic):
+    """The re-scaled median absolute deviation (sigma equivalent)."""
+    def value(self, values):
+        return mad_std(values)
+
+
+@dataclass(frozen=True)
+class Percentile(Statistic):
+    """An arbitrary percentile.
+
+    Parameters
+    ----------
+    percentile : `float`
+        A valid percentile (0 <= p <= 100).
+    """
+    percentile: float
+
+    def value(self, values):
+        return np.percentile(values, self.percentile)
+
+
+def compute_stats(values_ref, values_target, errors_target, row, stats, suffixes, prefix, skip_diff=False):
+    """Compute statistics on differences and store results in a row.
+
+    Parameters
+    ----------
+    values_ref : `numpy.ndarray`, (N,)
+        Reference values.
+    values_target : `numpy.ndarray`, (N,)
+        Measured values.
+    errors_target : `numpy.ndarray`, (N,)
+        Errors (standard deviations) on `values_target`.
+    row : `numpy.ndarray`, (1, C)
+        A numpy array with pre-assigned column names.
+    stats : `Dict` [`str`, `Statistic`]
+        A dict of `Statistic` values to measure, keyed by their column suffix.
+    suffixes : `Dict` [`str`, `Measurement`]
+        A dict of measurement types are the only valid values),
+        keyed by the column suffix.
+    prefix : `str`
+        A prefix for all column names (e.g. band).
+    skip_diff : `bool`
+        Whether to skip computing statistics on differences. Note that
+        differences will still be computed for chi statistics.
+
+    Returns
+    -------
+    row_stats : `numpy.ndarray`, (1, C)
```

Comment:
row_with_stats? Make it more obvious that this is the original row with the stats and not just the stats for the row.